internal/cmd/kafka/command_cluster_update.go

```diff
@@ -72,16 +73,20 @@ func (c *clusterCommand) update(cmd *cobra.Command, args []string, prompt form.P
 			return errors.New(errors.NonEmptyNameErrorMsg)
 		}
 		update.Spec.SetDisplayName(name)
-	} else {
-		update.Spec.SetDisplayName(*currentCluster.GetSpec().DisplayName)
 	}
 
-	updatedCku, err := c.validateResize(cmd, &currentCluster, prompt)
-	if err != nil {
-		return err
-	}
-	if updatedCku != -1 {
-		update.Spec.Config = &cmkv2.CmkV2ClusterSpecUpdateConfigOneOf{CmkV2Dedicated: &cmkv2.CmkV2Dedicated{Kind: "Dedicated", Cku: updatedCku}}
+	if cmd.Flags().Changed("cku") {
+		cku, err := cmd.Flags().GetUint32("cku")
+		if err != nil {
+			return err
+		}
+		updatedCku, err := c.validateResize(cmd, int32(cku), &currentCluster, prompt)
+		if err != nil {
+			return err
+		}
+		if updatedCku != -1 {
+			update.Spec.Config = &cmkv2.CmkV2ClusterSpecUpdateConfigOneOf{CmkV2Dedicated: &cmkv2.CmkV2Dedicated{Kind: "Dedicated", Cku: updatedCku}}
+		}
```

Comment:
No longer needed. The function will never return -1 without also returning an error.
```suggestion
		update.Spec.Config = &cmkv2.CmkV2ClusterSpecUpdateConfigOneOf{CmkV2Dedicated: &cmkv2.CmkV2Dedicated{Kind: "Dedicated", Cku: updatedCku}}
```